File: frontend/src/services/restaurantService.js
import api from './api';

export const restaurantService = {
  getAllRestaurants: () => api.get('/restaurants'),

  getRestaurantById: (id) => api.get(`/restaurants/${id}`),

  getRestaurantsByCity: (city) => api.get(`/restaurants/city/${city}`),

  createRestaurant: (data) => api.post('/restaurants', data),

  updateRestaurant: (id, data) => api.put(`/restaurants/${id}`, data),

  deleteRestaurant: (id) => api.delete(`/restaurants/${id}`),
};

File: src/main/java/com/example/demo/service/RestaurantService.java
package com.example.demo.service;

import com.example.demo.dto.request.RestaurantCreateRequestDTO;
import com.example.demo.dto.response.RestaurantDetailResponseDTO;
import java.util.List;

public interface RestaurantService {
    RestaurantDetailResponseDTO createRestaurant(RestaurantCreateRequestDTO request, Long ownerId);
    RestaurantDetailResponseDTO getRestaurantById(Long id);
    List<RestaurantDetailResponseDTO> getAllRestaurants();
    List<RestaurantDetailResponseDTO> getRestaurantsByCity(String city);
    List<RestaurantDetailResponseDTO> getOwnerRestaurants(Long ownerId);
    void updateRestaurant(Long id, RestaurantCreateRequestDTO request);
    void deleteRestaurant(Long id);
}